types/datum.go

```diff
@@ -1513,6 +1516,10 @@ func (d *Datum) ToMysqlJSON() (j json.BinaryJSON, err error) {
 		in = d.GetFloat64()
 	case KindMysqlDecimal:
 		in, err = d.GetMysqlDecimal().ToFloat64()
+		if err != nil {
```

Comment:
Not need to return here. It will return in line 1537.

Reply:
It not need to return in line 1533?

Reply:
you can add 
```
if err != nil {
   err = errros.Trace(err)
   return
}
```

at the end of the function.

Reply:
and line 1533 can be removed.